presto-tests/src/test/java/io/prestosql/tests/TestMetadataManager.java

```diff
@@ -151,4 +151,10 @@ public void testUpperCaseSchemaIsChangedToLowerCase()
                             return null;
                         });
     }
+
+    @Test
+    public void testUpperCaseListTablesFilter()
+    {
+        queryRunner.execute("SELECT * FROM system.jdbc.tables WHERE TABLE_SCHEM = 'UPPER_CASE_WHERE' AND TABLE_NAME = 'UPPER_CASE_WHERE'");
```

Comment:
Can you please validate the results? 

What should be the results here? Assuming https://github.com/prestosql/presto/issues/17, I guess we should return empty results here.

And so lowercasing in other place looks incorrect.

@findepi what are you thoughts?

Reply:
> What should be the results here? Assuming #17, I guess we should return empty results here.

Filter should behave the same regardless whether it's applied in engine, pushed down into connector  or applied "by hand by the user".
Since system.jdbc.tables does not contain any uppercase values in TABLE_NAME column, the `TABLE_NAME = 'UPPER_VALUE'` cannot possibly be true.
(until we introduce collations and make the TABLE_NAME use case insenstive collation)

Reply:
> And so lowercasing in other place looks incorrect.

which other place do you mean?

Reply:
@findepi 
So you are saying that 
`SELECT * FROM system.jdbc.tables WHERE TABLE_SCHEM = 'SOMEHITNG'`
should always return empty results, even if 'something' schema exists?

Reply:
That is correct.

Reply:
> which other place do you mean?

The one that was added in this pull request.

Reply:
I agree with @findepi, without collation servers strings should be compared as they're, without any modifications regardless the table/column.